v2/bolt_store.go

```diff
@@ -203,6 +204,16 @@ func (b *BoltStore) StoreLogs(logs []*raft.Log) error {
 
 	metrics.AddSample([]string{"raft", "boltdb", "logsPerBatch"}, float32(len(logs)))
 	metrics.AddSample([]string{"raft", "boltdb", "logBatchSize"}, float32(batchSize))
+	// Both the deferral and the inline function are important for this metrics
+	// accuracy. Deferral allows us to calculate the metric after the tx.Commit
+	// has finished and thus account for all the processing of the operation.
+	// The inlined function ensures that we do not calculate the time.Since(now)
+	// at the time of deferral but rather when the go runtime executes the
+	// deferred function.
```

Comment:
[nit/ learning for me] couldn't we  make this a func w some arguments like `time.now` from up top and the number of logs? and call it w `defer` ?

Reply:
We certainly could. Originally I had done that but then instead thought it might be nice to not have to follow fn calls to get to metrics that are emitted.

There are some extremely minor advantages/disadvantages for both but I don't have strong feelings about which way is best.

Reply:
Given how many args we would have to pass through I don't think its worth it here.